flyteplugins/go/tasks/pluginmachinery/internal/webapi/core.go

```diff
@@ -66,24 +67,39 @@ func (c CorePlugin) GetProperties() core.PluginProperties {
 }
 
 func (c CorePlugin) Handle(ctx context.Context, tCtx core.TaskExecutionContext) (core.Transition, error) {
```

Comment:
Maybe we can break this function into two, one handling sync and one async, for better readability?

Reply:
This is a good advice.
@pingsutw How do you think?
The implementation is comfortable for me.
Feel free to tell me the final decision, currently I think `break this function into two`  is more readable.